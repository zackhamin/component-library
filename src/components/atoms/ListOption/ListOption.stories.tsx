import { ListOption, ListOptionProps } from './ListOption';

export default {
	title: 'Atoms/ListOption',
	component: ListOption,
};

export const ListOptionStory = (args: ListOptionProps) => (
	<ListOption {...args} />
);

ListOptionStory.storyName = 'ListOption';
ListOptionStory.args = {
	children: (
		<>
			<h2>Perferendis eaque?</h2>
			<p>
				Lorem ipsum, dolor sit amet consectetur adipisicing elit. Voluptatum
				nihil laboriosam cumque nesciunt accusamus harum.
			</p>
		</>
	),
	isSelected: false,
};
ListOptionStory.argTypes = {
	handleSelect: { action: 'selected' },
};
