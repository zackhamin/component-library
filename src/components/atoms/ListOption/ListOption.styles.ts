import { colors } from 'Styles';
import styled from 'styled-components';

const StyledOption = styled.li`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	list-style: none;
	border: 1px solid ${colors.lightGrey};
	background: ${colors.white};
	border-radius: 8px;
	padding: 1.6rem;
	cursor: pointer;

	&.active {
		border: 1px solid ${colors.emerald50};
		background: rgba(3, 134, 120, 0.1);

		.radio {
			border: 1px solid ${colors.emerald};
		}

		.radio > span {
			background: ${colors.emerald};
		}
	}

	.details {
		margin-left: 2.6rem;
	}

	.radio {
		display: block;
		border: 1px solid ${colors.mediumGrey};
		border-radius: 50%;

		& > span {
			display: block;
			width: 10px;
			height: 10px;
			margin: 3px;
			border-radius: 50%;
			background: transparent;
		}
	}
`;

export { StyledOption };
