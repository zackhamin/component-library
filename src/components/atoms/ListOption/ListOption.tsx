import classNames from 'classnames';
import { ReactNode } from 'react';

import { StyledOption } from './ListOption.styles';

export interface ListOptionProps {
	children: ReactNode;
	isSelected: boolean;
	handleSelect: () => void;
}

export const ListOption = ({
	children,
	isSelected,
	handleSelect,
}: ListOptionProps) => {
	return (
		<StyledOption
			className={classNames({ active: isSelected })}
			onClick={handleSelect}
		>
			<div className="radio">
				<span />
			</div>
			<div className="details">{children}</div>
		</StyledOption>
	);
};
