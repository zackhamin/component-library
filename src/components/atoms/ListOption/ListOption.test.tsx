import { render } from '@testing-library/react';

import { ListOption } from './ListOption';

jest.spyOn(console, 'error').mockImplementation(jest.fn);

describe('ListOption', () => {
	it('renders correctly', () => {
		const { container } = render(
			<ListOption isSelected={false} handleSelect={jest.fn}>
				Some content
			</ListOption>
		);
		expect(container).toMatchSnapshot();
	});
});
