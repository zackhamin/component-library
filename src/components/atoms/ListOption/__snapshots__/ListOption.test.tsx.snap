// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOption renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style: none;
  border: 1px solid #EEEEEF;
  background: #ffffff;
  border-radius: 8px;
  padding: 1.6rem;
  cursor: pointer;
}

.c0.active {
  border: 1px solid #7FC1BA;
  background: rgba(3,134,120,0.1);
}

.c0.active .radio {
  border: 1px solid #00594C;
}

.c0.active .radio > span {
  background: #00594C;
}

.c0 .details {
  margin-left: 2.6rem;
}

.c0 .radio {
  display: block;
  border: 1px solid #bebebe;
  border-radius: 50%;
}

.c0 .radio > span {
  display: block;
  width: 10px;
  height: 10px;
  margin: 3px;
  border-radius: 50%;
  background: transparent;
}

<div>
  <li
    class="c0"
  >
    <div
      class="radio"
    >
      <span />
    </div>
    <div
      class="details"
    >
      Some content
    </div>
  </li>
</div>
`;
